framework/wazuh/core/wdb_http.py

```diff
@@ -216,6 +216,34 @@ async def set_agents_sync(self, agents_sync: dict) -> None:
         """
         await self._post('/agents/sync', agents_sync, empty_response=True)
 
+    async def get_agents_restart_info(self, ids: list, negate: bool) -> list[dict]:
+        """Retrieve restart information for agents.
+
+        Parameters
+        ----------
+        ids : list[int]
+            List of agent IDs to query
+        negate : bool
+            Whether to negate the ID filter (i.e., exclude these IDs).
+
+        Returns
+        -------
+        list[dict]
+            A list of agent restart information dicts.
+        """
+        request_data = {
+            "ids": [int(id) for id in ids],
+            "negate": negate
+        }
+
+        data = await self._post('/agents/restartinfo', request_data)
+        agents_restart_info = data.get('items', [])
+
+        for info in agents_restart_info:
+            info['id'] = f"{int(info['id']):03}"
```

Comment:
Perhaps a more readable way of achieving this would be

```suggestion
            info['id'] = str(info['id']).zfill(3)
```

Reply:
Fixed at https://github.com/wazuh/wazuh/pull/30505/commits/c07c045862c18e07cf7c296a18f585a4f77b4bbd